src/elevenlabs/conversational_ai/conversation.py

```diff
@@ -219,8 +226,46 @@ def _handle_message(self, message, ws):
             )
             if self.callback_latency_measurement and event["ping_ms"]:
                 self.callback_latency_measurement(int(event["ping_ms"]))
+        elif message["type"] == "client_tool_call":
+            event = message["client_tool_call_event"]
+            tool_name = event["tool_name"]
+            parameters = event["parameters"]
+            tool_call_id = event["tool_call_id"]
+            # Check if the tool exists in the configured client tools
+            if tool_name in self.config.client_tools:
+                try:
+                    # Execute the tool
+                    result = self.config.client_tools[tool_name](parameters)
+
+                    # Send back the result to the server
+                    ws.send(
+                        json.dumps({
+                            "type": "client_tool_result",
+                            "tool_call_id": tool_call_id,
+                            "result": result,
+                            "is_error": False,
+                        })
+                    )
+                except Exception as e:
+                    # Handle errors gracefully and send an error response
+                    ws.send(
+                        json.dumps({
+                            "type": "client_tool_result",
+                            "tool_call_id": tool_call_id,
+                            "result": f"Error: {str(e)}",
+                            "is_error": True,
+                        })
+                    )
         else:
```

Comment:
Isn't this indented wrong? This `else` is for the `message["type"]` if/elif chain, not for the `if tool_name in ...`

Reply:
Yes, fixing now